src/ts/components/settingspanel.ts

```diff
@@ -134,58 +136,61 @@ export class SettingsPanel extends Container<SettingsPanelConfig> {
 }
 
 /**
- * An item for a {@link SettingsPanel}, containing a {@link Label} and a component that configures a setting.
- * Supported setting components: {@link SelectBox}
+ * An item for a {@link SettingsPanel},
+ * If the components is a {@link SelectBox} it will handle the logic of displaying it or not
  */
 export class SettingsPanelItem extends Container<ContainerConfig> {
 
-  private label: Label<LabelConfig>;
-  private setting: SelectBox;
+  private setting: Component<ComponentConfig>;
 
   private settingsPanelItemEvents = {
     onActiveChanged: new EventDispatcher<SettingsPanelItem, NoArgs>(),
   };
 
-  constructor(label: string, selectBox: SelectBox, config: ContainerConfig = {}) {
+  constructor(setting: Component<ComponentConfig>, config: ContainerConfig = {}) {
```

Comment:
Keep the `label` as first parameter and don't render a label if `null` or `undefined` is passed in. Removing this parameter is a breaking API change that we do not want (some customers are using the framework so the API should be stable). Changing the second parameter from `SelectBox` to `Component` is ok because it widens the type and thus does not break the API.